cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/handler/StorageConsumptionCollectionHandler.java

```diff
@@ -3,46 +3,119 @@
         StorageConsumptionCollectionHandlerEvent data = event.getData();
 
         Consumption consumption = data.getContext().getConsumption();
         String environmentCrn = consumption.getEnvironmentCrn();
-        LOGGER.debug("Getting credential for environment with CRN [{}].", environmentCrn);
-        Credential credential = credentialService.getCredentialByEnvCrn(environmentCrn);
-        credentialConverter.convert(credential);
-
         Long resourceId = data.getResourceId();
         String resourceCrn = data.getResourceCrn();
         LOGGER.debug("Storage consumption collection started. resourceCrn: '{}'", resourceCrn);
-        return new SendStorageConsumptionEvent(SEND_CONSUMPTION_EVENT_EVENT.selector(), resourceId, resourceCrn, null);
+
+        try {
+            DetailedEnvironmentResponse detailedEnvironmentResponse = environmentService.getByCrn(environmentCrn);
+            LOGGER.debug("Getting credential for environment with CRN [{}].", environmentCrn);
+            String cloudPlatform = detailedEnvironmentResponse.getCloudPlatform();
+            if (cloudPlatform.equals(CloudPlatform.AWS.name())) {
+                Credential credential = credentialService.getCredentialByEnvCrn(environmentCrn);
+                CloudCredential cloudCredential = credentialConverter.convert(credential);
+
+                Date startTime = Date.from(Instant.now().minus(AMOUNT_TO_SUBTRACT, ChronoUnit.DAYS));
+                Date endTime = Date.from(Instant.now());
+                String bucketName = CloudStorageLocationUtil.getS3BucketName(consumption.getStorageLocation());
+                String region = detailedEnvironmentResponse.getLocation().getName();
+
+                ObjectStorageConnector connector = getObjectStorageConnector(cloudPlatform);
+                ObjectStorageSizeRequest request = ObjectStorageSizeRequest.builder()
+                        .withCredential(cloudCredential)
+                        .withRegion(Region.region(region))
+                        .withObjectStoragePath(bucketName)
+                        .withStartTime(startTime)
+                        .withEndTime(endTime)
+                        .build();
+                try {
+                    LOGGER.debug("Getting storage size for bucket {} and timeframe from {} to {}", bucketName, startTime, endTime);
+                    ObjectStorageSizeResponse response = connector.getObjectStorageSize(request);
+                    return new SendStorageConsumptionEvent(SEND_CONSUMPTION_EVENT_EVENT.selector(), resourceId, resourceCrn,
+                            new StorageConsumptionResult(response.getStorageInBytes()));
+                } catch (CloudConnectorException e) {
+                    LOGGER.error("Could not get storage size for bucket [{}]. Reason: {}}", bucketName, e.getMessage(), e);
+                    return new StorageConsumptionCollectionFailureEvent(resourceId, e, resourceCrn);
+                }
```

Comment:
Any particular reason to have a `try{}` within another `try{}`? Why not add a `catch (CloudConnectorException e) {}` to the outer `try{}` instead?

Reply:
It was only because if the catch was mowed to the outer try then the `bucketName` wouldn't be available to log, because it's inside the if branch. But I can omit that from the log and move the catch out if it makes the function cleaner.